### PR TITLE
statedb: Fix WriteJSON with multiple tables

### PR DESCRIPTION
**Pull request by joamaki**
WriteJSON produced broken JSON with multiple tables. This refactors it to use a buffered writer and extends the tests to check that it works with two tables instead of just one.

### CODE REVIEW

File: pkg/statedb/db.go [hunk 1]
```diff
@@ -50,38 +51,38 @@ var _ DB = &stateDB{}
...
+			first = false
+		}
 		iter, err := tx.Get(table, "id")
 		if err != nil {
 			return err
 		}
-		if _, err := w.Write([]byte("\"" + table + "\": [\n")); err != nil {
-			return err
-		}
+		buf.WriteString("\"" + table + "\": [\n")
```
Comment (bimmlerd) [inline pkg/statedb/db.go:67]: Bit weird to ignore errors in some places but not in other places - either throw it everywhere or ignore it, I guess?

**Review by bimmlerd — APPROVED** [review]
generally seems fine, would be good to clean up the error handling

File: pkg/statedb/db.go (hunk 1 above)
Comment (joamaki) [reply pkg/statedb/db.go:67]: The buffered writer carries the error over multiple calls and I only need to care about the error on `Flush()`. Removed the unnecessary checks.

Comment (bimmlerd) [reply pkg/statedb/db.go:67]: oh yeah, good to know - I guess it's not really part of the documented API but I doubt that it'll ever change.

**Comment by joamaki** [comment]
> generally seems fine, would be good to clean up the error handling

Cleaned up the error handling.

**Review by ti-mo — APPROVED** [review]

**Pull request merged**